run_tests.py

```diff
@@ -53,117 +59,185 @@ def modified_environment(**kwargs):
-    parser.add_argument('-p', '--ioc-path', default=None, help="The path to the folder containing the IOC's st.cmd")
-    parser.add_argument('-e', '--emulator-path', default=None, help="The path of the lewis.py file")
-    parser.add_argument('-py', '--python-path', default="C:\Instrument\Apps\Python\python.exe", help="The path of python.exe")
-    parser.add_argument('-ep', '--emulator-protocol', default='stream', help="The Lewis protocal to use (optional)")
-    parser.add_argument('-r', '--record-simulation', default=False, action="count",
-                        help="Use record simulation rather than emulation (optional)")
-    parser.add_argument('-ea', '--emulator-add-path', default=None, help="Add path where device packages exist for the emulator.")
-    parser.add_argument('-ek', '--emulator-device-package', default=None, help="Name of packages where devices are found.")
-
-    parser.add_argument('--var-dir', default=None, help="Directory in which to create a log dir to write log file to and "
-                                                        "directory in which to create tmp dir which contains environments "
-                                                        "variables for the IOC. Defaults to environment variable ICPVARDIR and current dir if empty.")
+    parser.add_argument('-l', '--list-devices',
+                        help="List available devices for testing.", action="store_true")
+    parser.add_argument('-pf', '--prefix', default=os.environ.get("MYPVPREFIX", None),
+                        help='The instrument prefix; e.g. TE:NDW1373')
+    parser.add_argument('-tm', '--test-module', default=None, nargs="+",
+                        help="Test module to run")
+    parser.add_argument('-e', '--emulator-path', default=emulator_path,
+                        help="The path of the lewis.py file")
+    parser.add_argument('-py', '--python-path', default="C:\Instrument\Apps\Python\python.exe",
+                        help="The path of python.exe")
+    parser.add_argument('--var-dir', default=None,
+                        help="Directory in which to create a log dir to write log file to and directory in which to "
+                             "create tmp dir which contains environments variables for the IOC. Defaults to "
+                             "environment variable ICPVARDIR and current dir if empty.")
 
     arguments = parser.parse_args()
 
     if arguments.list_devices:
         print("Available tests:")
         print('\n'.join(package_contents("tests")))
-        exit(0)
+        sys.exit(0)
+
+    var_dir = arguments.var_dir if arguments.var_dir is not None else os.getenv("ICPVARDIR", os.curdir)
+
+    if arguments.prefix is None:
+        print("Cannot run without instrument prefix")
+        sys.exit(-1)
+
+    if arguments.emulator_path is None:
+        print("Cannot run without emulator path")
+        sys.exit(-1)
+
+    module_results = []
+
+    if arguments.test_module is not None:
```

Comment:
This is a bit copy-and-paste can you join the test_module and package_contents lists then loop through them together?